artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/codec/PageCountRecordInc.java

```diff
@@ -26,17 +26,20 @@
 
    private int value;
 
+   private long persistentSize;
```

Comment:
Seems my comment is removed due to update. Should be -1.

Reason for this is that if it isn't set e.g. have read up an old journal as such decode wont set it, and you know you need to calculate it to get a correct value.

Reply:
@michaelandrepearce  - That is fine I will make those values -1, but I don't think we would try and re-calculate a correct value.  When loading the journal you'd have to re-load the messages to calculate from the paging store which we don't want to do.  I think the proper behavior is to just ignore it and not use the value in the calculations.  Then over time as new records come in the values would be filled in and correct.  

If I change the value to -1 on the record, are you thinking we should just have the getter do the negative check and return 0 if it's -1 or were you thinking to just return -1 from the getter and then check for negative everywhere else the value is used?